File: se.softhouse.garden.orchid.demo.vaadin/src/main/java/se/softhouse/garden/orchid/demo/vaadin/DemoApplication.java
package se.softhouse.garden.orchid.demo.vaadin;

import javax.annotation.Resource;

import se.softhouse.garden.orchid.spring.DI;
import se.softhouse.garden.orchid.spring.text.OrchidLocalizedMesageSource;

import com.vaadin.Application;
import com.vaadin.ui.Window;

@SuppressWarnings("serial")
public class DemoApplication extends Application {

	@SuppressWarnings("unused")
	private final Void di = DI.inject(this);

	@Resource
	protected OrchidLocalizedMesageSource msg;

	@Override
	public void init() {
		this.msg.setLocale(getLocale());
		Window mainWindow = new Window(MSG.WINDOW_TITLE);
		setMainWindow(mainWindow);
		mainWindow.setSizeFull();
		mainWindow.addComponent(new DemoView());
	}
}

File: se.softhouse.garden.orchid.scala.demo.vaadin/src/main/scala/se/softhouse/garden/orchid/scala/demo/vaadin/DemoApplication.scala
package se.softhouse.garden.orchid.scala.demo.vaadin

import javax.annotation.Resource
import se.softhouse.garden.orchid.scala.spring.DI
import se.softhouse.garden.orchid.scala.spring.OrchidMessage.OrchidMessageToString
import se.softhouse.garden.orchid.spring.text.OrchidLocalizedMesageSource
import vaadin.scala.Application

class DemoApplication extends {
  @transient @Resource val msgs: OrchidLocalizedMesageSource = null
} with Application with DI {
  override def main = {
    msgs.setLocale(getLocale())
    mainWindow.caption = MSG.WINDOW_TITLE
    new DemoView()
  }
}
